src/main/java/io/frictionlessdata/tableschema/field/DateField.java

```diff
@@ -33,23 +33,22 @@ public DateField(String name, String format, String title, String description,
     public LocalDate parseValue(String value, String format, Map<String, Object> options)
             throws InvalidCastException, ConstraintsException {
 
-        Pattern pattern = Pattern.compile(REGEX_DATE);
-        Matcher matcher = pattern.matcher(value);
+        if(value.length()>patternFormat.length() || value.length() < "y-M-d".length()){
```

Comment:
Like for the YearField, I believe this check is superfluous. The matcher would throw anyways for a String like `9-1-2`, right?

Reply:
same as for YearField, I'll roll it back